src/message_interface/message_handler.rs

```diff
@@ -79,13 +79,20 @@ impl ClientMessageHandler {
         let _ = message.response_tx.send(Response::new(message.message_type, response));
     }
 
+    // secret_manager needs to be mutable for Stronghold
+    #[allow(unused_mut)]
     async fn call_client_method(&self, method: &ClientMethod) -> Result<ResponseType> {
         match method {
             ClientMethod::GenerateAddresses {
                 secret_manager,
                 options,
             } => {
-                let secret_manager: SecretManager = secret_manager.parse()?;
+                let mut secret_manager: SecretManager = secret_manager.parse()?;
+                // If we use stronghold we need to read the snapshot in case it hasn't been done already
+                #[cfg(feature = "stronghold")]
+                if let SecretManager::Stronghold(stronghold_secret_manager) = &mut secret_manager {
+                    stronghold_secret_manager.read_stronghold_snapshot().await?;
+                }
```

Comment:
As said on Discord, these repeated blocks can probably be abstracted away somehow. Just commenting to open an issue from it for later.

Reply:
https://github.com/iotaledger/iota.rs/issues/973